src/Model/Table/QueuedJobsTable.php

```diff
@@ -563,7 +563,33 @@ public function reset($id = null) {
 			$conditions['id'] = $id;
 		}
 
-		return (bool)$this->updateAll($fields, $conditions);
+		return $this->updateAll($fields, $conditions);
+	}
+
+	/**
+	 * @param string $type
+	 * @param string|null $reference
+	 *
+	 * @return int
+	 */
+	public function rerun($type, $reference = null) {
+		$fields = [
+			'completed' => null,
+			'fetched' => null,
+			'progress' => 0,
+			'failed' => 0,
+			'workerkey' => null,
+			'failure_message' => null,
+		];
+		$conditions = [
+			'completed IS NOT' => null,
+			'job_type' => $type,
```

Comment:
Optionally an index here would help massive updates. I don't think we have a solid use case however

Reply:
Yeah, this is not the case for many queue use cases out there.